Generic API for monitoring LwM2M FOTA libray (#10176)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

* Updated:

  * Use defines from the :ref:`lib_nrf_cloud` library for nRF Cloud related string values.

>   * Integrated new LwM2M client utils FOTA callback functionality

Reviewer: ```suggestion
  * The application now integrates the :ref:`lib_lwm2m_client_utils` FOTA callback functionality.
```

    * Use defines from the :ref:`lib_nrf_cloud` library for nRF Cloud related string values.
      Remove the inclusion of the file :file:`nrf_cloud_codec.h`.

* :ref:`lwm2m_client` sample:

  * Updated:

>     * Integrated new LwM2M client utils FOTA callback functionality

Reviewer 2: `Integrated new LwM2M client utils FOTA callback`

>   * Updated:
>     * Integrated new LwM2M client utils FOTA callback functionality

Reviewer: ```suggestion
  * Updated:
  
    * The sample now integrates the :ref:`lib_lwm2m_client_utils` FOTA callback functionality.
```

    * A public header file :file:`nrf_cloud_defs.h` that contains common defines for interacting with nRF Cloud and the :ref:`lib_nrf_cloud` library.

> * LwM2M client utils library:

Reviewer: ```suggestion
* :ref:`lib_lwm2m_client_utils` library:
```

  * Updated:

    * :file:`lwm2m_client_utils.h` Fota API for give function callback for indicate
      state changes and request permission for start Update process.

>     * Removed old API lwm2m_firmware_get_update_state_cb().

Reviewer 2: ```
    * :file:`lwm2m_client_utils.h` New API for FOTA to register application callback to receive
      state changes and requests for update process.
    * Removed old API lwm2m_firmware_get_update_state_cb().
```

>     * :file:`lwm2m_client_utils.h` Fota API for give function callback for indicate
>       state changes and request permission for start Update process.
>     * Removed old API lwm2m_firmware_get_update_state_cb().

Reviewer: @SeppoTakalo, a slightly changed version on top of your suggestion:
```suggestion
    * :file:`lwm2m_client_utils.h` includes new API for FOTA to register application callback to receive state changes and requests for update process.
      
  * Removed:
  
    * The old API ``lwm2m_firmware_get_update_state_cb()``.
```
